src/librustc_driver/driver.rs

```diff
@@ -1247,7 +1249,9 @@ pub fn collect_crate_types(session: &Session, attrs: &[ast::Attribute]) -> Vec<c
 }
 
 pub fn compute_crate_disambiguator(session: &Session) -> String {
-    let mut hasher = Sha256::new();
+    use std::hash::Hasher;
+
+    let mut hasher = ArchIndependentHasher::new(Blake2bHasher::new(256 / 8, &[]));
```

Comment:
There's no real reason other than that we had a 256 bit hash here. 128 bits would probably be more than enough.